### ISSUE TITLE
only getting compute errors

### ISSUE THREAD

**Issue by gabcraftia** [xmrig/xmrig-amd#156]
if i start the miner i only get job received and get compute errors dont get any shares only errors
i am using a rx 480 with the latest drivers
someone has a solution?

**Comment by calvintam236** [comment]
i'm getting same problem with vega 56. @xmrig can you help?

**Comment by xmrig** [comment]
Don't use latest drivers
1. #146
2. https://github.com/fireice-uk/xmr-stak/issues/1797

**Comment by gabcraftia** [comment]
I use the blockchain drivers and het the problem

**Comment by Festour** [comment]
> Don't use latest drivers


Wich one we should use then? Can you write number of latest version that still works well?

**Comment by xmrig** [comment]
> Wich one we should use then? Can you write number of latest version that still works well?

18.3.4, 18.5.1, 18.5.2, 18.6.1.

**Comment by JungleCatSW** [comment]
I have been testing the the code for the monero algo against the moneroworld pool. It has been working well. However I did have some occasional compute errors, is the new algo affected by this AMD driver bug at all.

**Comment by bdmayes** [comment]
So are we basically waiting on AMD to resolve this issue then, or is there anything the mining software can do? I using the AMD pro drivers on Ubuntu 18.04, by following this guide:

https://github.com/xmrminer01102018/VegaToolsNConfigs/blob/master/VegaUbuntuGuide

I'll admit it's not the easiest guide to follow, but the TL;DR is basically:
- install 18.10 drivers with PAL open CL
- modify a few of the sources files slightly (such as using cmath instead of math headers)
- build xmrig-amd
- uninstall 18.10 drivers and install 18.30 AMD pro drivers instead (to allow for some additional features)
- Reboot and now you can mine with xmrig-amd binary that was compiled against the 18.10 drivers, but using the 18.30 drivers to get better performance


This has been working fine for a few months, but as of yesterday, my mining pool started giving me a "warning" that I need to update to 2.8.0 (for upcoming PoW change). Even though this is supposed to be a warning, I can no longer mine at all -- I just get "n/a" for all of my vega56 cards. I tried rolling back to 18.10 drivers, pulling the latest xmrig-amd from git, and then recompiling. Then I reinstall 18.30 drivers and now I just get thread compute errors.

I'm a little bit lost on exactly what the problem is here, and whether or not there is any end in sight. Any further updates would be appreciated.

**Comment by xmrig** [comment]
@bdmayes I add option to restore old cache behavior https://github.com/xmrig/xmrig-amd/commit/462ea5893c3c112386053ebcd365ed77da66a560 now instead of `cmake ..` you should run `cmake .. -DSTRICT_CACHE=OFF`. Also all changes related to cmath/math not required anymore.

Now it in dev branch, it will be included to next stable release v2.8.1 around October 9-12.
Thank you.

**Comment by bdmayes** [comment]
@xmrig Are you saying that the latest code will now build properly on the 18.30 drivers if we simply use `cmake .. -DSTRICT_CACHE=OFF`? If so, I will gladly try that tonight. Not having to change math headers to cmath would also be nice. There were also changes from `isnormal` to `std::isnormal`. Was that also fixed?

**Comment by xmrig** [comment]
No `-DSTRICT_CACHE=OFF` just allow use same hack with 2 driver versions as before, still need use 18.10 first.
`isnormal` fixed too.

**Comment by xmrig** [comment]
Issue with invalid shares may fixed, reference https://github.com/fireice-uk/xmr-stak/pull/1866
This change added to dev branch too, please check.

**Comment by bdmayes** [comment]
I'm not using xmr-stak at all, only xmrig-amd. However, I don't know if that fix is relevant or not as it specifically references the rocm open cl implementation and I am using pal. I will certainly pull the latest commit from this repo (xmrig-amd) tonight, build it on 18.10 with `-DSTRICT_CACHE=OFF`, and then reinstall 18.30 drivers and retry.

I should hopefully have an update in about 8 hours or so.

**Comment by bdmayes** [comment]
I'm not using xmr-stak at all, only xmrig-amd. I do see your commit on the dev branch, so I'll pull from that instead of master and give it a shot. However, I don't know if that fix is relevant or not as it specifically references the rocm open cl implementation on xmr-stak#1866 and I am using pal (AMD Pro drivers). I will certainly pull the latest commit from the dev branch on this repo (xmrig-amd) tonight, build it on 18.10 with `-DSTRICT_CACHE=OFF`, and then reinstall 18.30 drivers and retry.

I should hopefully have an update in about 7-8 hours or so.

**Comment by unsivilaudio** [comment]
@bdmayes Your issue is actually with the 18.30 drivers. For whatever reason the opencl cache files are incorrectly generated on 18.30. Presumably you tested your miner on 18.10 and pre-generated your cache files before moving to 18.30 so the miner did not have to regenerate them until recently. Any worksize change will require a new cache file to be created. I also run a Vega rig on Linux and only by chance did I figure this out. Using Ubuntu 18.04 LTS which is incompatible with the 18.10 driver, I copied my cache files from an Ethos installation I was testing xmrig on.

This development is very interesting though and I also may have to pull and compile it to test.

**Comment by Spudz76** [comment]
The definitive thread on the bug is [here](https://github.com/fireice-uk/xmr-stak/issues/1797)

The gstoner posts mostly are the good parts.  It's a mismatch of whether the shader C code (mostly same code between all CN OpenCL miners) works properly when the middle representation is HSAIL/PAL/SPIR or whatever the compiler within the driver uses, which got changed in both AMDGPU-Pro and ROCm around the same time, and now ROCm also has at least two variants PAL and whatever the other one is.

**Comment by bdmayes** [comment]
@unsivilaudio I think you nailed it! I am BACK! I don't login as root, but I have to launch my miner with sudo privileges, and I remember being unable to remove the entire directory (before a fresh `git clone`) because of those .cache files. So I removed it with sudo. And you're correct, I definitely tested it on 18.10 at a lower hash rate before moving on to 18.30. So here is what I just did:

- uninstalled 18.30 drivers and reboot
- install 18.10 drivers (`./amdgpu-pro-install -y --opencl=pal`) and reboot
- fresh `git clone` of xmrig-amd and then `git checkout dev`
- Ran the following to build the miner:
```
$ git clone https://github.com/xmrig/xmrig-amd.git
$ cd xmrig-amd
$ cd src/3rdparty
$ mv CL CL_original
$ ln -s /usr/include/CL CL
$ cd ../..
$ mkdir build
$ cd build
$ cmake .. -DSTRICT_CACHE=OFF
$ make
```
- Test it on 18.10 and generate the cache files with `sudo ./xmrig-amd`. This only yields about 1.2 KH/s for each card.
- Uninstall the 18.10 drivers with `sudo /usr/bin/amdgpu-pro-uninstall`
- reboot
- Install the 18.30 drivers with `./amdgpu-pro-install -y --opencl=pal`
- If you haven't already, then add `amdgpu.ppfeaturemask=0xffffffff` to your /etc/default/grub file just after `quiet splash` 
- `sudo update-grub`
- reboot


Now you should be able to do whatever setup you want to do on your cards (I set fan speed, overclock, and set power play tables to lower wattage). Run the miner and it should work again, getting around 1.9 KH/s per card. 😁 

```
$ sudo ./xmrig-amd
 * ABOUT        XMRig-AMD/2.8.1-dev gcc/7.3.0
 * LIBS         libuv/1.18.0 OpenCL/2.0 OpenSSL/1.1.0g microhttpd/0.9.59
 * CPU          Intel(R) Celeron(R) CPU G3900 @ 2.80GHz x64 AES
 * ALGO         cryptonight, donate=1%
 * POOL #1      gulf.moneroocean.stream:10001 variant 1
 * COMMANDS     hashrate, pause, resume
[2018-10-05 22:08:21] compiling code and initializing GPUs. This will take a while...
[2018-10-05 22:08:21] #0, GPU #0 Radeon RX Vega, intensity: 896 (8/256), unroll: 8, cu: 56
[2018-10-05 22:08:21] #1, GPU #0 Radeon RX Vega, intensity: 896 (8/256), unroll: 8, cu: 56
[2018-10-05 22:08:21] #2, GPU #0 Radeon RX Vega, intensity: 896 (8/256), unroll: 8, cu: 56
[2018-10-05 22:08:21] #3, GPU #0 Radeon RX Vega, intensity: 896 (8/256), unroll: 8, cu: 56
[2018-10-05 22:08:21] #4, GPU #1 Radeon RX Vega, intensity: 896 (8/256), unroll: 8, cu: 56
[2018-10-05 22:08:21] #5, GPU #1 Radeon RX Vega, intensity: 896 (8/256), unroll: 8, cu: 56
[2018-10-05 22:08:21] #6, GPU #1 Radeon RX Vega, intensity: 896 (8/256), unroll: 8, cu: 56
[2018-10-05 22:08:21] #7, GPU #1 Radeon RX Vega, intensity: 896 (8/256), unroll: 8, cu: 56
...
[2018-10-05 22:17:32] accepted (78/0) diff 150557 (1409 ms)
| THREAD | GPU | 10s H/s | 60s H/s | 15m H/s |
|      0 |   0 |   474.3 |   468.0 |     n/a |
|      1 |   0 |   473.2 |   471.2 |     n/a |
|      2 |   0 |   477.0 |   468.7 |     n/a |
|      3 |   0 |   461.7 |   465.0 |     n/a |
|      4 |   1 |   478.4 |   478.2 |     n/a |
|      5 |   1 |   478.3 |   477.1 |     n/a |
|      6 |   1 |   481.1 |   480.9 |     n/a |
|      7 |   1 |   478.4 |   477.3 |     n/a |
[2018-10-05 22:17:32] speed 10s/60s/15m 3802.8 3786.8 n/a H/s max 3818.8 H/s
```

**Comment by bdmayes** [comment]
Just FYI, I tried removing the third card entirely and I'm still getting compute errors. I can try lowering the intensity, but these settings have been stable for me without a single compute error on my rig for about a month. Now that I have removed the 3rd card, the only change is that I updated the code to the latest dev branch. :(

**Comment by unsivilaudio** [comment]
You're going to need to restore your cache file. I hope you backed it up or your going to have to downgrade your driver to recreate it again.

**Comment by bdmayes** [comment]
@unsivilaudio Maybe I don't quite follow then. I already downgraded to 18.10 last night and tested the miner, which generated the cache file. After getting it all back up and running on 18.30 I simply powered down the machine, installed the third card, and then started the miner back up. Are you saying that each time I add a new card, I have to downgrade to 18.10, regenerate the cache files, then go back up to 18.30?

**Comment by bdmayes** [comment]
Ok that might have been the problem. I did the downgrade, cache recreate, and then upgrade. It has been almost an hour and so far, no compute errors. I'm going to let it run overnight and see what it shows in the morning. It seems that my setup is considerably more finicky than I realized it would be.

I'll report back if there are further errors in the morning, but it seems stable thus far. This cache file things seems incredibly important. I'm surprised your comment yesterday was the first I've ever read about it. Thanks again!

**Comment by unsivilaudio** [comment]
It gets corrupted on occasion, not sure what causes it but occasionally I
have to delete the file and restore my backup.

**Comment by bdmayes** [comment]
Looks like I have been stable throughout the night. It has been running for almost 8 hours without a single error. I guess I'll setup a cron job to periodically copy the cache elsewhere for safe keeping.

**Comment by fernandodrf** [comment]
Hi there, I am using version 2.8.1 and I just getting compute errors.

My rig:

Gigabyte AX370 Gaming 5
Ryzen 1600
Ubuntu 18.04 LTS 64bit
Two Radeon RX 480 8Gb
AMD driver: 18.30-641594 

I tried to compile it with the DSTRICT_CACHE=OFF flag but it makes no difference. Do I have to downgrade the AMD drivers?

Thanks a lot !

**Comment by xmrig** [comment]
`-DSTRICT_CACHE=OFF` only allow use cache created with older drivers, so need downgrade, run miner to create cache with old driver, delete old driver then install new driver.

**Comment by emailyc** [comment]
is there an update to usable drivers?

**Comment by micotito** [comment]
@bdmayes I compiled 2.8.2-dev with -DSTRICT_CACHE=OFF with 18.10 drivers and tested and created cache file all OK. Then removed 18.10 and installed 18.30 drivers reboot. but when xmrig-amd starts it compiles cache again and get 3 Thread Compute Errors and then hangs until reboot. Any idea.? Have you tried 2.8.2

**Comment by bdmayes** [comment]
@micotito See the thread I referenced above. Unfortunately, even when I pull down the latest code and recompile, I can't get anything to work. It seems like things are completely unusable since yesterday's fork. 😢

**Comment by xmrig** [comment]
v2.8.4 in dev branch now correctly works with 18.10 <-> 18.30 driver switching, short manual https://github.com/xmrig/xmrig-amd/blob/master/doc/DRIVERS.md I checked it and got a little better hashrate compared to Windows. In addition autoconfig for Vega now should work well.

@xmrminer01102018 Good job, but all `cmath/math`, `std::isnormal/isnormal` stuff not required at all, miner successfully compiled without these patches. Also don't need replace OpenCL headers, generic headers is good enough.

**Comment by bdmayes** [comment]
@xmrig I'm still unable to get any hashrate reported. I do like the new output with the cache file being printed out though. Very useful!

- Do I need to update anything in config.json after the Oct 18 fork?
- Maybe I need to try a different pool? Perhaps MoneroOcean is somehow incompatible?

```
$ sudo ./xmrig-amd
 * ABOUT        XMRig-AMD/2.8.4-dev gcc/7.3.0
 * LIBS         libuv/1.18.0 OpenCL/2.0 OpenSSL/1.1.0g 
 * CPU          Intel(R) Celeron(R) CPU G3900 @ 2.80GHz x64 AES
 * ALGO         cryptonight, donate=1%
 * POOL #1      gulf.moneroocean.stream:10001 variant 1
 * COMMANDS     hashrate, pause, resume
[2018-10-20 08:35:55] compiling code and initializing GPUs. This will take a while...
[2018-10-20 08:35:56] #0, GPU #0 Radeon RX Vega, intensity: 896 (8/256), unroll: 8, cu: 56
[2018-10-20 08:35:56]            CACHE: ./.cache/AG46HF5IVUPMUCKLQ7Y2NEECATXWOOK6O2KT72VP6QT4XWNFLXWQ.bin
[2018-10-20 08:35:56] GPU #0 compiling...
[2018-10-20 08:36:05] GPU #0 compilation completed, elapsed time 8.84s
[2018-10-20 08:36:05] #1, GPU #0 Radeon RX Vega, intensity: 896 (8/256), unroll: 8, cu: 56
[2018-10-20 08:36:05]            CACHE: ./.cache/AG46HF5IVUPMUCKLQ7Y2NEECATXWOOK6O2KT72VP6QT4XWNFLXWQ.bin
[2018-10-20 08:36:05] #2, GPU #0 Radeon RX Vega, intensity: 896 (8/256), unroll: 8, cu: 56
[2018-10-20 08:36:05]            CACHE: ./.cache/AG46HF5IVUPMUCKLQ7Y2NEECATXWOOK6O2KT72VP6QT4XWNFLXWQ.bin
[2018-10-20 08:36:05] #3, GPU #0 Radeon RX Vega, intensity: 896 (8/256), unroll: 8, cu: 56
[2018-10-20 08:36:05]            CACHE: ./.cache/AG46HF5IVUPMUCKLQ7Y2NEECATXWOOK6O2KT72VP6QT4XWNFLXWQ.bin
[2018-10-20 08:36:05] #4, GPU #1 Radeon RX Vega, intensity: 896 (8/256), unroll: 8, cu: 56
[2018-10-20 08:36:05]            CACHE: ./.cache/AG46HF5IVUPMUCKLQ7Y2NEECATXWOOK6O2KT72VP6QT4XWNFLXWQ.bin
[2018-10-20 08:36:05] #5, GPU #1 Radeon RX Vega, intensity: 896 (8/256), unroll: 8, cu: 56
[2018-10-20 08:36:05]            CACHE: ./.cache/AG46HF5IVUPMUCKLQ7Y2NEECATXWOOK6O2KT72VP6QT4XWNFLXWQ.bin
[2018-10-20 08:36:05] #6, GPU #1 Radeon RX Vega, intensity: 896 (8/256), unroll: 8, cu: 56
[2018-10-20 08:36:05]            CACHE: ./.cache/AG46HF5IVUPMUCKLQ7Y2NEECATXWOOK6O2KT72VP6QT4XWNFLXWQ.bin
[2018-10-20 08:36:05] #7, GPU #1 Radeon RX Vega, intensity: 896 (8/256), unroll: 8, cu: 56
[2018-10-20 08:36:05]            CACHE: ./.cache/AG46HF5IVUPMUCKLQ7Y2NEECATXWOOK6O2KT72VP6QT4XWNFLXWQ.bin
[2018-10-20 08:36:05] #8, GPU #2 Radeon RX Vega, intensity: 896 (8/256), unroll: 8, cu: 56
[2018-10-20 08:36:05]            CACHE: ./.cache/AG46HF5IVUPMUCKLQ7Y2NEECATXWOOK6O2KT72VP6QT4XWNFLXWQ.bin
[2018-10-20 08:36:05] #9, GPU #2 Radeon RX Vega, intensity: 896 (8/256), unroll: 8, cu: 56
[2018-10-20 08:36:05]            CACHE: ./.cache/AG46HF5IVUPMUCKLQ7Y2NEECATXWOOK6O2KT72VP6QT4XWNFLXWQ.bin
[2018-10-20 08:36:05] #10, GPU #2 Radeon RX Vega, intensity: 896 (8/256), unroll: 8, cu: 56
[2018-10-20 08:36:05]            CACHE: ./.cache/AG46HF5IVUPMUCKLQ7Y2NEECATXWOOK6O2KT72VP6QT4XWNFLXWQ.bin
[2018-10-20 08:36:05] #11, GPU #2 Radeon RX Vega, intensity: 896 (8/256), unroll: 8, cu: 56
[2018-10-20 08:36:05]            CACHE: ./.cache/AG46HF5IVUPMUCKLQ7Y2NEECATXWOOK6O2KT72VP6QT4XWNFLXWQ.bin
[2018-10-20 08:36:05] use pool gulf.moneroocean.stream:10001  45.32.81.217 
[2018-10-20 08:36:05] new job from gulf.moneroocean.stream:10001 diff 1000 algo cn/2
[2018-10-20 08:36:11] accepted (1/0) diff 1000 (80 ms)
[2018-10-20 08:36:14] accepted (2/0) diff 1000 (57 ms)
[2018-10-20 08:36:14] accepted (3/0) diff 1000 (93 ms)
[2018-10-20 08:36:14] accepted (4/0) diff 1000 (150 ms)
[2018-10-20 08:36:17] accepted (5/0) diff 1000 (57 ms)
[2018-10-20 08:36:17] accepted (6/0) diff 1000 (113 ms)
[2018-10-20 08:36:18] accepted (7/0) diff 1000 (161 ms)
[2018-10-20 08:36:30] speed 10s/60s/15m n/a n/a n/a H/s max n/a H/s
| THREAD | GPU | 10s H/s | 60s H/s | 15m H/s |
|      0 |   0 |     n/a |     n/a |     n/a |
|      1 |   0 |     n/a |     n/a |     n/a |
|      2 |   0 |     n/a |     n/a |     n/a |
|      3 |   0 |     n/a |     n/a |     n/a |
|      4 |   1 |     n/a |     n/a |     n/a |
|      5 |   1 |     n/a |     n/a |     n/a |
|      6 |   1 |     n/a |     n/a |     n/a |
|      7 |   1 |     n/a |     n/a |     n/a |
|      8 |   2 |     n/a |     n/a |     n/a |
|      9 |   2 |     n/a |     n/a |     n/a |
|     10 |   2 |     n/a |     n/a |     n/a |
|     11 |   2 |     n/a |     n/a |     n/a |
[2018-10-20 08:36:35] speed 10s/60s/15m n/a n/a n/a H/s max n/a H/s
[2018-10-20 08:36:35] accepted (20/0) diff 1000 (57 ms)
[2018-10-20 08:36:37] accepted (21/0) diff 1000 (57 ms)
[2018-10-20 08:36:38] accepted (22/0) diff 1000 (113 ms)
[2018-10-20 08:36:38] accepted (23/0) diff 1000 (57 ms)
[2018-10-20 08:36:38] accepted (24/0) diff 1000 (113 ms)
[2018-10-20 08:36:38] accepted (25/0) diff 1000 (113 ms)
[2018-10-20 08:36:40] accepted (26/0) diff 1000 (57 ms)
[2018-10-20 08:36:40] accepted (27/0) diff 1000 (113 ms)
[2018-10-20 08:36:41] accepted (28/0) diff 1000 (57 ms)
[2018-10-20 08:36:41] accepted (29/0) diff 1000 (57 ms)
[2018-10-20 08:36:42] new job from gulf.moneroocean.stream:10001 diff 23424 algo cn/2
| THREAD | GPU | 10s H/s | 60s H/s | 15m H/s |
|      0 |   0 |     n/a |     n/a |     n/a |
|      1 |   0 |     n/a |     n/a |     n/a |
|      2 |   0 |     n/a |     n/a |     n/a |
|      3 |   0 |     n/a |     n/a |     n/a |
|      4 |   1 |     n/a |     n/a |     n/a |
|      5 |   1 |     n/a |     n/a |     n/a |
|      6 |   1 |     n/a |     n/a |     n/a |
|      7 |   1 |     n/a |     n/a |     n/a |
|      8 |   2 |     n/a |     n/a |     n/a |
|      9 |   2 |     n/a |     n/a |     n/a |
|     10 |   2 |     n/a |     n/a |     n/a |
|     11 |   2 |     n/a |     n/a |     n/a |
[2018-10-20 08:36:44] speed 10s/60s/15m n/a n/a n/a H/s max n/a H/s
[2018-10-20 08:36:48] accepted (30/0) diff 1000 (81 ms)
[2018-10-20 08:36:48] accepted (31/0) diff 1000 (57 ms)
[2018-10-20 08:36:48] accepted (32/0) diff 1000 (57 ms)
[2018-10-20 08:36:49] accepted (33/0) diff 1000 (57 ms)
[2018-10-20 08:36:49] accepted (34/0) diff 1000 (113 ms)
[2018-10-20 08:36:51] accepted (35/0) diff 1000 (57 ms)
[2018-10-20 08:36:51] accepted (36/0) diff 1000 (113 ms)
[2018-10-20 08:36:51] accepted (37/0) diff 1000 (113 ms)
[2018-10-20 08:36:53] accepted (38/0) diff 1000 (81 ms)
[2018-10-20 08:36:59] speed 10s/60s/15m n/a n/a n/a H/s max n/a H/s
```

**Comment by bdmayes** [comment]
nanopool seems to report a hash rate, but every share is rejected due to "low difficulty"

```
[2018-10-20 08:42:10] speed 10s/60s/15m 3546.6 n/a n/a H/s max 3013.6 H/s
[2018-10-20 08:42:11] rejected (0/1) diff 120001 "Low difficulty share" (227 ms)
| THREAD | GPU | 10s H/s | 60s H/s | 15m H/s |
|      0 |   0 |   200.2 |     n/a |     n/a |
|      1 |   0 |   391.1 |     n/a |     n/a |
|      2 |   0 |   388.8 |     n/a |     n/a |
|      3 |   0 |   198.0 |     n/a |     n/a |
|      4 |   1 |   292.9 |     n/a |     n/a |
|      5 |   1 |   286.7 |     n/a |     n/a |
|      6 |   1 |   289.1 |     n/a |     n/a |
|      7 |   1 |   280.7 |     n/a |     n/a |
|      8 |   2 |   297.6 |     n/a |     n/a |
|      9 |   2 |   301.9 |     n/a |     n/a |
|     10 |   2 |   306.8 |     n/a |     n/a |
|     11 |   2 |   307.9 |     n/a |     n/a |
[2018-10-20 08:42:23] speed 10s/60s/15m 3542.0 n/a n/a H/s max 3547.5 H/s
| THREAD | GPU | 10s H/s | 60s H/s | 15m H/s |
|      0 |   0 |   197.6 |     n/a |     n/a |
|      1 |   0 |   390.1 |     n/a |     n/a |
|      2 |   0 |   390.1 |     n/a |     n/a |
|      3 |   0 |   198.4 |     n/a |     n/a |
|      4 |   1 |   289.6 |     n/a |     n/a |
|      5 |   1 |   286.8 |     n/a |     n/a |
|      6 |   1 |   286.8 |     n/a |     n/a |
|      7 |   1 |   283.6 |     n/a |     n/a |
|      8 |   2 |   300.9 |     n/a |     n/a |
|      9 |   2 |   303.9 |     n/a |     n/a |
|     10 |   2 |   302.0 |     n/a |     n/a |
|     11 |   2 |   306.7 |     n/a |     n/a |
[2018-10-20 08:42:38] speed 10s/60s/15m 3537.2 n/a n/a H/s max 3547.5 H/s
[2018-10-20 08:42:50] speed 10s/60s/15m 3542.7 693.0 n/a H/s max 3547.5 H/s
[2018-10-20 08:42:54] new job from xmr-us-west1.nanopool.org:14444 diff 120001 algo cn/1
| THREAD | GPU | 10s H/s | 60s H/s | 15m H/s |
|      0 |   0 |   195.3 |   197.0 |     n/a |
|      1 |   0 |   390.6 |   389.8 |     n/a |
|      2 |   0 |   403.3 |   394.3 |     n/a |
|      3 |   0 |   195.5 |   196.9 |     n/a |
|      4 |   1 |   287.8 |   289.5 |     n/a |
|      5 |   1 |   288.0 |   287.5 |     n/a |
|      6 |   1 |   288.1 |   287.6 |     n/a |
|      7 |   1 |   286.8 |   284.5 |     n/a |
|      8 |   2 |   304.6 |   301.4 |     n/a |
|      9 |   2 |   303.7 |   303.9 |     n/a |
|     10 |   2 |   304.2 |   304.1 |     n/a |
|     11 |   2 |   302.2 |   305.9 |     n/a |
[2018-10-20 08:43:10] speed 10s/60s/15m 3550.6 3542.8 n/a H/s max 3550.5 H/s
[2018-10-20 08:43:41] rejected (0/2) diff 120001 "Low difficulty share" (228 ms)
```

**Comment by xmrig** [comment]
Yes you need update a lot of things, so better use default config.json and fill required fields to add your pool, miner will create threads with proper settings, it easy way, hard way below:

* Change `"variant"` option on each pool to `-1`.
* Change `"strided_index"` on each thread to `2`, it reason for very low/unreported hashrate.
* Increase `"worksize"` to `16`.
* Increase `"intensity"` to `1920` (good start value, best value may differ) it valid for all cryptonight/* algorithms.

**Comment by bdmayes** [comment]
@xmrig I just made those changes but I'm getting the following now. I'm not sure what it means:

```
[2018-10-20 09:24:25] new job from gulf.moneroocean.stream:10001 diff 1000 algo cn/2
[2018-10-20 09:24:30] Error CL_MEM_OBJECT_ALLOCATION_FAILURE when calling clEnqueueNDRangeKernel for kernel 0.
[2018-10-20 09:24:30] Error CL_MEM_OBJECT_ALLOCATION_FAILURE when calling clEnqueueNDRangeKernel for kernel 0.
[2018-10-20 09:24:34] Error CL_MEM_OBJECT_ALLOCATION_FAILURE when calling clEnqueueNDRangeKernel for kernel 0.
[2018-10-20 09:24:35] Error CL_MEM_OBJECT_ALLOCATION_FAILURE when calling clEnqueueNDRangeKernel for kernel 0.
```


Perhaps I need to make a symbolic link the CL headers instead of using the defaults?

**Comment by unsivilaudio** [comment]
Yeah 1920 is for 2 threads, you literally are trying to use more memory than you have available.  Try 896 if you are sticking with 4 threads.

**Comment by bdmayes** [comment]
I see. Sorry -- I just copied this config from a friend that had things working several months ago. I have no idea if it's better to have 2 threads per GPU or 4. I just tried changing my config to 2 threads with the values above and it seemed to just hang my entire system. I had to just hold the power button to turn it off. So I just went back for 4 threads with 896 intensity, and it start working with 18.10 drivers. :tada:

I'm going to reinstall 18.30 and see how it goes. Will report back shortly.

**Comment by unsivilaudio** [comment]
I personally like 4 threads at 896 (these intensities are only applicable on 2mb scratchpad aka regular cryptonight v0/1/2; cn-lite has 1mb scratchpad, and cn-heavy is 4mb).

**Comment by xmrig** [comment]
My bad, 1920 is for double threads.

**Comment by bdmayes** [comment]
Well it seems to be working with the 4 thread solution again on 2.8.4. You are both amazingly helpful. Thank you so much. The only downside is that my rig is pulling more watts now, and my hashrate is down a bit. The new fork is cutting into my profitability. :cry:

I'm back up and running. I believe the reason was that I needed to update variant and strided_index for each thread in the config. Thank you both again!

```
[2018-10-20 09:55:44] new job from gulf.moneroocean.stream:10001 diff 211741 algo cn/2
| THREAD | GPU | 10s H/s | 60s H/s | 15m H/s |
|      0 |   0 |   415.5 |   414.6 |     n/a |
|      1 |   0 |   418.8 |   413.3 |     n/a |
|      2 |   0 |   418.4 |   417.6 |     n/a |
|      3 |   0 |   413.7 |   415.4 |     n/a |
|      4 |   1 |   397.5 |   396.6 |     n/a |
|      5 |   1 |   398.2 |   396.7 |     n/a |
|      6 |   1 |   397.5 |   401.1 |     n/a |
|      7 |   1 |   397.6 |   391.6 |     n/a |
|      8 |   2 |   424.8 |   422.4 |     n/a |
|      9 |   2 |   425.3 |   415.5 |     n/a |
|     10 |   2 |   425.7 |   421.1 |     n/a |
|     11 |   2 |   425.7 |   420.5 |     n/a |
[2018-10-20 09:55:48] speed 10s/60s/15m 4959.0 4926.8 n/a H/s max 4962.4 H/s
```

**Comment by unsivilaudio** [comment]
tbh I haven't messed with cn/2 on my vega rig.  However I think you can gain some by keeping worksize at 8 and lowering unroll to 4 or 2, but my understanding of these settings is still rudimentary; please experiment. Its weird how Polaris only observes about a 2% drop in hashrate.

**Comment by bdmayes** [comment]
I'm not choosing cn/2 -- my mining pool is as it switches algorithms. I believe there is a way to configure the miner to only accept certain algorithms, and ignore others? I just haven't looked into how to set that up yet, especially because I was getting 5.7KH/s at around 520W prior to the fork. Right now I'm getting 5 KH/s and pulling 595W off the wall.

**Comment by xmrminer01102018** [comment]
I just finished reading this whole thread.  @xmrig  - Thanks for fixing the code. @bdmayes - Nice that you have cnv2 working without errors.  I just need to test and update my guide. @xmrig - If you haven't figured it out yet I'm user20180110 on hashvault Discord.

**Comment by knittycatkitty** [comment]
this is great development, i made some startup scripts in case anyone needs them

followed your notes xmrminer01102018 !

**Comment by knittycatkitty** [comment]
no matter which method i do for some reason xmrig and xmr stak both keep recompiling 

i saved the cache files generated from 18.10 drivers, tarred those up

, untarred them after installing 18.30

still recompiling on 18.30 

any suggestions?

im using cast for the meantime !!

**Comment by TomskDiver** [comment]
I use blockchain drivers and mining be OK. Now, i install 18.6.1 driver and get THREAD #0 COMPUTE ERROR:
```
 * ABOUT        XMRig-AMD/2.8.5 MSVC/2017
 * LIBS         libuv/1.23.1 OpenCL/2.0 OpenSSL/1.1.1 microhttpd/0.9.59
 * CPU          Intel(R) Core(TM) i5-4690 CPU @ 3.50GHz x64 AES
 * ALGO         cryptonight, donate=5%
 * POOL #1      xmr.pool.minergate.com:45700 variant 2
 * COMMANDS     hashrate, pause, resume
[2018-11-14 09:44:08] compiling code and initializing GPUs. This will take a while...
[2018-11-14 09:44:08] found AMD platform index: 1, name: Advanced Micro Devices, Inc.
[2018-11-14 09:44:08] #0, GPU #0 AMD Radeon HD 7800 Series, intensity: 256 (8/256), unroll: 8, cu: 16
[2018-11-14 09:44:08] GPU #0 compiling...
[2018-11-14 09:44:18] GPU #0 compilation completed, elapsed time 9.76s
[2018-11-14 09:44:18] use pool xmr.pool.minergate.com:45700  136.243.102.157
[2018-11-14 09:44:18] new job from xmr.pool.minergate.com:45700 diff 1063 algo cn/2
[2018-11-14 09:44:24] accepted (1/0) diff 1063 (111 ms)
[2018-11-14 09:44:26] THREAD #0 COMPUTE ERROR
[2018-11-14 09:44:34] accepted (2/0) diff 1063 (102 ms)
[2018-11-14 09:44:38] THREAD #0 COMPUTE ERROR
[2018-11-14 09:44:45] THREAD #0 COMPUTE ERROR
[2018-11-14 09:44:51] accepted (3/0) diff 1063 (99 ms)
[2018-11-14 09:44:53] accepted (4/0) diff 1063 (97 ms)
[2018-11-14 09:45:06] accepted (5/0) diff 1063 (96 ms)
[2018-11-14 09:45:11] speed 10s/60s/15m 126.5 n/a n/a H/s max 126.4 H/s
[2018-11-14 09:45:12] accepted (6/0) diff 1063 (101 ms)
[2018-11-14 09:45:24] accepted (7/0) diff 1063 (96 ms)
[2018-11-14 09:45:32] accepted (8/0) diff 1063 (97 ms)
[2018-11-14 09:45:42] new job from xmr.pool.minergate.com:45700 diff 1017 algo cn/2
[2018-11-14 09:45:53] accepted (9/0) diff 1017 (113 ms)
[2018-11-14 09:46:11] speed 10s/60s/15m 126.9 126.5 n/a H/s max 126.9 H/s
[2018-11-14 09:46:21] THREAD #0 COMPUTE ERROR
[2018-11-14 09:46:27] accepted (10/0) diff 1017 (96 ms)
```

**Comment by xmrminer01102018** [comment]
@knittycatkitty -As of right now once you have compiled the code with older driver, you cannot change the config.json settings.  If you do, it will recompile.  When it does, you will get compute errors.  So the best route is 1. Tweak config.json in 18.30 and ignore compute errors.  2. Once you get the best hash rate, uninstall 18.30 and install 18.10. 3. Compile xmrig-amd with "cmake -DSTRICT_CACHE=OFF ..". 4. Uninstall 18.10 and reinstall 18.30.  5. Reboot, update and run the miner.

**Comment by bdmayes** [comment]
@xmrminer01102018 Any thoughts on my issues described in https://github.com/xmrig/xmrig-amd/issues/180 ? Specifically, the last post there is the most relevant. The PPT definitely seems to be the problem. I just ran the following steps, and still one card has entirely n/a threads:

- Hooked up all 4 of my cards to the rig
- Uninstalled the 18.40 drivers and reboot
- Installed 18.10 drivers and reboot
- Did a `git pull` on the dev branch to get the latest code
- `rm -rf build && mkdir build && cd build && cmake .. -DSTRICT_CACHE=OFF && make`
- Added the 4th card into config.json
- deleted the current .cache/*.bin file
- ran xmrig-amd to generate the new cache file
- uninstalled 18.10 and reboot
- reinstalled 18.30 (instead of 18.40 -- though it doesn't seem to many any difference) and reboot
- run xmrig-amd and all 4 cards work just fine, but I'm drawing 985W from the wall

Now it's working at least but the power consumption is very high and I want to tune it to help lower some of the power draw. So then I tried:

- I set PPT on all 4 cards, and card 0 got n/a for all of its threads. Cannot even gracefully shutdown the computer with `sudo reboot` -- instead I have to hold the power button for 5 seconds.
- powered back on and set PPT on cards 1, 2, and 3 (but _not_ 0). Unfortunately card 0 still shows n/a threads.
- press and hold power button, then power back on
- run xmrig-amd with any PPT changes and all 4 cards work just fine


Any idea why PPT applied to one card can affect a totally different card?

**Comment by Spudz76** [comment]
[AMD says PowerPlay is (still) broken in newer drivers](https://github.com/RadeonOpenCompute/ROCm/issues/433)

That also explains as best I've seen anywhere, how the AMD triple-layer-stack ends up so broken all the time.  Three groups do different stuff without apparently communicating with each other very often.

**Comment by bdmayes** [comment]
@Spudz76 Thanks for that link. Finally an explanation!

For what it's worth, I stumbled upon the ability to set power levels in `/sys/class/drm/card*/device/hwmon/hwmon*/power1_cap` files and I'm getting some really great results. Seems like the sweet spot is somewhere between 120 and 125W. I shaved off 300W at the wall for a trivial drop in hashrate. So far so good so I guess I'll run with this for now and wait to see if PPT ever gets fixed.

**Comment by xmrminer01102018** [comment]
@bdmayes -  I have seen this on what I categorized as lower grade memory state V56.  It can be caused by mostly inferior memory or bad riser.  I use the different PPT file for that situation.  If you want to try that they are in PPTDIR folder at my GitHub site.  I named it LGMV56PPT.

**Comment by xmrig** [comment]
https://github.com/xmrig/xmrig/releases/tag/v4.0.0-beta
